### ISSUE TITLE
[no-throw-literal] Resolve ESLint known limitations

### ISSUE THREAD

**Issue by julienfouilhe** [typescript-eslint/typescript-eslint#1258]
**Repro**

```JSON
{
  "rules": {
    "no-throw-literal": "error"
  }
}
```

```TS
const str = 'Error';

throw str;
```

**Expected Result**

`throw-no-literal` should report an error

**Actual Result**

`throw-no-literal` does not report an error

**Additional Info**

It is a known ESLint limitation, listed [here](https://eslint.org/docs/rules/no-throw-literal), but can't we leverage typescript's typesystem and fix those with a `@typescript-eslint/no-throw-literal` rule?
